### PR TITLE
chore(deps): Upgrade Rust to 1.62.0

### LINKED ISSUE #13474: Upgrade Rust to 1.62.0
https://blog.rust-lang.org/2022/06/30/Rust-1.62.0.html

### PR DESCRIPTION
**Pull request by jszwedko**
Closes #13474

### CODE REVIEW

File: lib/vector-common/src/conversion/tests/unix.rs [hunk 1]
```diff
@@ -55,6 +55,7 @@ fn dateref() -> DateTime<Utc> {
     Utc.from_utc_datetime(&NaiveDateTime::from_timestamp(981_173_106, 0))
 }
 
+#[allow(clippy::trait_duplication_in_bounds)] // appears to be a false positive
```
Comment (jszwedko) [inline lib/vector-common/src/conversion/tests/unix.rs:58]: Same here.

File: src/sources/util/unix_stream.rs [hunk 2]
```diff
@@ -133,9 +133,6 @@ pub fn build_unix_stream_source(
             );
         }
 
-        // Cleanup
-        drop(stream);
```
Comment (jszwedko) [inline src/sources/util/unix_stream.rs:137]: Apparently `stream` didn't implement `Drop` so dropping it here had no real effect.

File: lib/vector-common/src/conversion.rs [hunk 3]
```diff
@@ -141,6 +141,7 @@ impl Conversion {
     /// # Errors
     ///
     /// Returns errors from the underlying conversion functions. See `enum Error`.
+    #[allow(clippy::trait_duplication_in_bounds)] // appears to be a false positive
```
Comment (jszwedko) [inline lib/vector-common/src/conversion.rs:144]: Asking in Slack about this one before I create a clippy issue.

Comment (spencergilbert) [reply lib/vector-common/src/conversion.rs:144]: Planning on merging regardless?

File: src/config/unit_test/mod.rs [hunk 4]
```diff
@@ -53,7 +53,7 @@ impl UnitTest {
         let (topology, _) = topology::start_validated(self.config, diff, self.pieces)
             .await
             .unwrap();
-        let _ = topology.sources_finished().await;
```
Comment (spencergilbert) [inline src/config/unit_test/mod.rs:56]: These always bothered me, glad it got cleaned up!

**Review by spencergilbert — APPROVED** [review]

**Review by bruceg — APPROVED** [review]
I'm a little bothered by all the `.expect`s on string writes, but otherwise LGTM

**Comment by Salathaniel** [comment]
https://e.vg/SO72S? sign up to get a 10 US Dollar newcomer bonus! I made
more than 200 US Dollars by watching the video here, so you can try it.

**Review by neuronull — APPROVED** [review]
Rust upgrades aren't something dependabot can detect or, something we intentionally exclude ?

**Comment by bruceg** [comment]
> Rust upgrades aren't something dependabot can detect or, something we intentionally exclude ?

No, I don't believe dependabot is set up to handle compiler versioning. We don't obviously have it intentionally excluded. That upgrade is frequently not something dependabot can easily produce a automated PR for.

**Pull request merged**